hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1531,6 +1533,13 @@ def _composite(f):
         )
     if params[0].default is not sig.empty:
         raise InvalidArgument("A default value for initial argument will never be used")
+    if not is_func_param_called_within(f, params[0].name):
+        # raise DeprecationWarning("First parameter should be referenced within")
+        return note_deprecation(
+            "First parameter should be referenced within",
+            since="2022-07-16",
```

Comment:
```suggestion
            "There is no reason to use @st.composite on a function which "
            "does not call the provided draw() function internally.",
            since="RELEASEDAY",
```